iothub/service/src/Common/Exceptions/DeviceAlreadyExistsException.cs

```diff
@@ -24,6 +24,11 @@ public DeviceAlreadyExistsException(string message, Exception innerException)
         {
         }
 
+        public DeviceAlreadyExistsException(ErrorCode code, string message, Exception innerException = null)
```

Comment:
This class is public, so wondering if you should just create a new constructor? I'm not sure if this will be considered a breaking change if someone has taken dependency on this class (though unlikely).